crypto/merklekeystore/PersistentKeystore.go

```
@@ -68,87 +83,56 @@ func (p *PersistentKeystore) Persist(keys []crypto.GenericSigningKey, firstValid
 	}
 
-	key := &crypto.GenericSigningKey{}
 	err = protocol.Decode(keyB, key)
 	if err != nil {
-		return nil, fmt.Errorf("PersistentKeystore.GetKey: %w", err)
-	}
-
-	return key, nil
-}
-
-// DropKeys deletes the keys up to the specified round (including)
-func (p *PersistentKeystore) DropKeys(round uint64) (count int64, err error) {
-	err = p.store.Atomic(func(ctx context.Context, tx *sql.Tx) error {
-		res, err := tx.Exec("DELETE FROM StateProofKeys WHERE round <= ?", round)
-		if err == nil {
-			count, err = res.RowsAffected()
-		}
-		return err
-	})
-	return count, err
-}
-
-func keystoreInstallDatabase(tx *sql.Tx) error {
-	_, err := tx.Exec(`CREATE TABLE StateProofKeys (
-    	id	  INTEGER PRIMARY KEY, 
-    	round INTEGER,	    --*  committed round for this key
-		key   BLOB  --*  msgpack encoding of ParticipationAccount.StateProof.GenericSigningKey
-		);`)
-	if err != nil {
-		return err
-	}
-
-	_, err = tx.Exec(`CREATE UNIQUE INDEX IF NOT EXISTS roundIdx ON StateProofKeys (round);`)
-	if err != nil {
-		return err
+		return nil, 0, err // fmt.Errorf("PersistentKeystore.GetKey: %w", err)
 	}
-	_, err = tx.Exec("INSERT INTO schema (tablename, version) VALUES (?, ?)", keystoreTableSchemaName, keystoreSchemaVersion)
 
-	return err
+	return key, round, nil
 }
 
-// RestoreKeystore loads the PersistentKeystore from given database
-func RestoreKeystore(store db.Accessor) (PersistentKeystore, error) {
+// CountKeys counts the number of rows in StateProofKeys table
+func (s *Keystore) CountKeys(store db.Accessor) int {
+	var count int
 	err := store.Atomic(func(ctx context.Context, tx *sql.Tx) error {
-		return MigrateDB(tx)
+		row := tx.QueryRow("SELECT COUNT(*) FROM StateProofKeys")
+		err := row.Scan(&count)
+		if err != nil {
+			return fmt.Errorf("failed to count rows in table StateProofKeys : %w", err)
+		}
+		return nil
 	})
 	if err != nil {
-		return PersistentKeystore{}, err
+		return 0
```

I think we should return an error here and stop the migration process in the caller.